models/classes/runner/QtiRunnerService.php

```diff
@@ -1506,38 +1506,42 @@ protected function onTimeout(RunnerServiceContext $context, AssessmentTestSessio
         $this->getServiceManager()->get(EventManager::SERVICE_ID)->trigger($event);
 
         $isLinear = $session->getCurrentNavigationMode() === NavigationMode::LINEAR;
+        $logContext = [];
+        if ($context instanceof QtiRunnerServiceContext) {
+            $logContext['deliveryExecutionId'] = $context->getTestExecutionUri();
+        }
         switch ($timeOutException->getCode()) {
             case AssessmentTestSessionException::ASSESSMENT_TEST_DURATION_OVERFLOW:
-                \common_Logger::i('TIMEOUT: closing the assessment test session');
+                \common_Logger::i('TIMEOUT: closing the assessment test session', $logContext);
```

Comment:
WDYT 
   If create an internal class inside that repo that overload common_Logger
   where we could set the QtiRunnerServiceContext object 
   send all messages through it 
   
   It will probably create more flexibility if we decide to extend context data for all that places in future

Reply:
I see that you didn't do it for some reason just for my self clearification

Reply:
You still have to have access to `RunnerServiceContext` there, which you will probably pass from the outside, given the static nature of the logger.
I don't see how it could help much.